src/intervals/arithmetic/power.jl

```diff
@@ -292,40 +271,42 @@ function log1p(a::F) where {T, F<:Interval{T}}
 end
 
 """
-    nthroot(a::Interval{BigFloat}, n::Integer)
+    nthroot(a::Interval, n::Integer)
 
 Compute the real n-th root of Interval.
 """
-function nthroot(a::Interval{BigFloat}, n::Integer)
+function nthroot(a::F, n::Integer) where {F<:Interval{BigFloat}}
+    isempty(a) && return a
     n == 1 && return a
     n == 2 && return sqrt(a)
-    n < 0 && isthinzero(a) && return emptyinterval(a)
-    isempty(a) && return a
-    if n > 0
-        alo = inf(a)
-        ahi = sup(a)
-        ahi < 0 && iseven(n) && return emptyinterval(BigFloat)
-        if alo < 0 && ahi >= 0 && iseven(n)
-            a = a ∩ Interval{BigFloat}(0, Inf)
-        end
-        ui = convert(Culong, n)
-        low = BigFloat()
-        high = BigFloat()
-        ccall((:mpfr_rootn_ui, :libmpfr), Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , low , a.lo , ui, MPFRRoundDown)
-        ccall((:mpfr_rootn_ui, :libmpfr), Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , high , a.hi , ui, MPFRRoundUp)
-        b = interval(low , high)
-        return b
-    elseif n < 0
-        return inv(nthroot(a, -n))
-    elseif n == 0
-        return emptyinterval(a)
+    n == 0 && return emptyinterval(a)
+    # n < 0 && isthinzero(a) && return emptyinterval(a)
+    n < 0 && return inv(nthroot(a, -n))
+
+    alo, ahi = bounds(a)
+    ahi < 0 && iseven(n) && return emptyinterval(BigFloat)
+    if alo < 0 && ahi >= 0 && iseven(n)
+        a = a ∩ F(0, Inf)
+        alo, ahi = bounds(a)
     end
+    ui = convert(Culong, n)
+    low = BigFloat()
+    high = BigFloat()
+    ccall((:mpfr_rootn_ui, :libmpfr), Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , low , alo , ui, MPFRRoundDown)
+    ccall((:mpfr_rootn_ui, :libmpfr), Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , high , ahi , ui, MPFRRoundUp)
+    return interval(low , high)
 end
 
-function nthroot(a::Interval{T}, n::Integer) where T
+function nthroot(a::F, n::Integer) where {T, F<:Interval{T}}
     n == 1 && return a
     n == 2 && return sqrt(a)
-    abig = @biginterval(a)
+
+    abig = bigequiv(a)
+    if n < 0
+        issubnormal(mag(a)) && return inv(nthroot(a, -n))
```

Comment:
The short answer is because that's the only way I was able to make that some tests in power_rev finally passed. 😬

The deeper reason, I presume, is related to the conversion of a subnormal number to a BigFloat (as a side note, `BigFloat`/MPFR does not have subnormal numbers). When you convert `5.0e-324` (`nextfloat(0.0)`, which is the upper boundary of some of the intervals in the tests that now pass) to a `BigFloat` (using `bigequiv`, for 53-bits of bigfloat precision), strangely to me, you get: 4.9406564584124654e-324 (more precision does not help here., or changes the argument). This is strange to me because I was expecting exactly `5.0e-324`; it certainly might well be the usual issue of the floating point representation, though I think that subnormal numbers are exactly representable. So I simply did not rely on the BigFloat conversion...

Reply:
Back to the failing/passing test, this is what we have now:
```julia
julia> a = 0 .. nextfloat(0.0); 

julia> b = Interval(1.5367463556376293e46, Inf); # the "correct" answer according to ITF1788

julia> n = -7
-7

julia> inv(nthroot(a, -n)) # current implementation; agrees with b
Interval(1.5367463556376293e46, Inf)

julia> inv(nthroot(IA.@biginterval(a), -n)) # using Interval{BigFloat}
Interval(1.53674635563762978699561409504936312650237729242224250716551531177882837331969e+46, Inf)

julia> Interval{Float64}(ans) # conversion to Interval{Float64}; tighter than `b`
Interval(1.5367463556376296e46, Inf)
```
So the implementation with `bigequiv` seems better, but again, do we mean literally `5.0e-324` or the actual `nextfloat(0.0)`...?

Reply:
> fwiw removing the check all tests still pass in IA

Bit not in IC...